protocols/luci-proto-wireguard/root/usr/libexec/rpcd/luci.wireguard

```diff
@@ -15,6 +15,14 @@ local methods = {
 			return {keys = {priv = prv, pub = pub}}
 		end
 	},
+	getPublicKeyFromPrivate = {
+		args = {privkey = "privkey"},
+		call = function(args)
+			local pubkey = sys.exec("echo '" .. args.privkey .. "' | wg pubkey 2>/dev/null"):sub(1, -2)
+
+			return {keys = {priv = args.privkey, pub = pubkey}}
```

Comment:
I don't think you need the private key to be returned here. Wouldn't it be better to only return the public key?

Reply:
This function is extendable, and can be co-opted for other purposes later, when the private key is (also) desirable. This also means we can perform a sanity check whereby we verify what we sent to the lower layers. The very creation of this new function is ample reason why both should be included.